README.md

```diff
@@ -73,12 +73,62 @@ const collectionAssetsResponse = await client.listAssets({
 });
 ```
 
+### Generating Stark (Layer 2) keys
+
+Stark keys are required by users in order to transact on ImmutableX's StarkEx Layer 2. They are the equivalent of Ethereum keys on L1, and allows users to sign transactions, and receive and store tokens.
+
+#### Key registration
+
+On ImmutableX, the goal of generating a Stark key is to [register](https://docs.x.immutable.com/docs/how-to-register-users/) a mapping between this key and the user's Ethereum key so that transactions requiring both L1 and L2 signers can be executed by users.
+
+#### How to generate Stark keys on ImmutableX
+
+ImmutableX provides two Stark key generation methods:
+| Type of Stark key generated | How do applications connect to these Stark accounts? | Factors determining when this method should be used: | ImmutableX tools: |
+| --- | --- | --- | --- |
+| [Deterministic](#generating-or-retrieving-a-deterministic-key) - generated using the user's Ethereum key as a seed (which means that the same Ethereum key will always generate the same Stark key) | Users connect with their L1 wallet (ie. Metamask), as the L2 key can simply be obtained from the L1 key. | ***User experience*** - do you want the user to have full responsibility for storing their L2 key?<br/><br/> ***Interoperability*** - how do most applications connect to user's Stark wallets? If a user's key is randomly generated, then they will not be able to connect to an application that requires them to sign in with their L1 key to generate their L2 key.  | [Link SDK](https://docs.x.immutable.com/docs/link-setup)<br/><br/>Core SDK's [`generateLegacyStarkPrivateKey()`](https://github.com/immutable/imx-core-sdk/blob/main/src/utils/stark/starkCurve.ts#L152) method |
```

Comment:
For the deterministic option, the user isn't responsible for storing their L2 key - the Link SDK deterministically generates it

Reply:
Interoperability - the Link SDK is widely used by marketplaces and DAPPs to let users connect with ImmutableX L2. Generating the key deterministically will ensure users can connect and transact with all the marketplaces and DAPPs that implement the Link SDK.